From 5e79a54c265d6ef195d02cbcb525b10049fe68cd Mon Sep 17 00:00:00 2001
Message-Id: <5e79a54c265d6ef195d02cbcb525b10049fe68cd.1413837856.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: Adam Rutkowski <adam.j.rutkowski@intel.com>
Date: Fri, 14 Feb 2014 19:58:12 +0100
Subject: [PATCH 060/312] FOR UPSTREAM [VPG]: drm/i915: Fix for i915_timestamp

Current implementation of i915_timestamp in debugfs uses unsupported
arithmetic operations to convert ftrace timestamp, causing build error
for 32-bit architectures.  This patch is fixing this by switching to
'do_div' macro.

Issue: VIZ-3488
Change-Id: Ia3f1a4cb1b76c559ce91dec8f217a219891e0233
Depends-Change-Id: I408d64af344462786aa919caed3d80d25a123312
Signed-off-by: Adam Rutkowski <adam.j.rutkowski@intel.com>
Reviewed-on: https://android.intel.com/163895
Tested-by: sys_abtbuild <sys_abtbuild@intel.com>
Reviewed-by: Madajczak, Tomasz <Tomasz.Madajczak@intel.com>
Tested-by: Madajczak, Tomasz <Tomasz.Madajczak@intel.com>
Reviewed-by: Yang, Fei <fei.yang@intel.com>
Reviewed-by: Bloomfield, Jon <jon.bloomfield@intel.com>
Reviewed-by: Barbalho, Rafael <rafael.barbalho@intel.com>
Reviewed-by: Odeberg, Hans <hans.odeberg@intel.com>
---
 drivers/gpu/drm/i915/i915_debugfs.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/i915/i915_debugfs.c b/drivers/gpu/drm/i915/i915_debugfs.c
index 66abbee..30f9e5c 100644
--- a/drivers/gpu/drm/i915/i915_debugfs.c
+++ b/drivers/gpu/drm/i915/i915_debugfs.c
@@ -4109,8 +4109,8 @@ ssize_t i915_timestamp_read(struct file *filp,
 	ftrace_ts = ftrace_now(cpu);
 	local_irq_restore(flags);
 
-	sec  = (u32)(ftrace_ts / NSEC_PER_SEC);
-	nsec = (u32)(ftrace_ts % NSEC_PER_SEC);
+	nsec = do_div(ftrace_ts, NSEC_PER_SEC);
+	sec = (u32) ftrace_ts;
 
 	len = snprintf(buf, TIMESTAMP_BUFFER_LEN,
 		      "CPU%03u %u.%09u s\nGPU %u ticks\n",
-- 
1.7.9.5

